types/react-redux/index.d.ts

```diff
@@ -46,7 +50,7 @@ interface ComponentMergeDecorator<TMergedProps, TOwnProps> {
  * @param mergeProps
  * @param options
  */
-export declare function connect(): ComponentDecorator<DispatchProp<any>, {}>;
+export declare function connect(): ComponentDecoratorInfer<DispatchProp<any>>;
```

Comment:
No problem. Hopefully all bases are covered now - this change was mostly in the opposite direction making only the single endpoint inferred rather than introducing something new.